analyses/oncoprint-landscape/01-plot-oncoprint.R

```diff
@@ -177,6 +178,50 @@ metadata <- metadata %>%
   # Reorder display_group based on display_order
   dplyr::mutate(display_group = forcats::fct_reorder(display_group, display_order))
 
+# Filter to the metadata associated with the broad histology value, if provided
+if (!is.null(opt$broad_histology)) {
+  
+  if (!opt$broad_histology == "Other CNS") {
+    metadata <- metadata %>%
+      dplyr::filter(broad_histology == opt$broad_histology)
+  } else {
+    metadata <- metadata %>%
+      dplyr::filter(
+        broad_histology %in% c(
+          "Tumors of sellar region",
+          "Neuronal and mixed neuronal-glial tumor",
+          "Tumor of cranial and paraspinal nerves	",
+          "Meningioma",
+          "Mesenchymal non-meningothelial tumor	",
+          "Germ cell tumor",
+          "Choroid plexus tumor	",
+          "Histiocytic tumor",
+          "Tumor of pineal region",
+          "Metastatic tumors",
+          "Other astrocytic tumor",
+          "Lymphoma",
+          "Melanocytic tumor",
+          "Other tumor"
+        )
+      )
+  }
+  
+  # There's a duplicated `Tumor_Sample_Barcode` column that we will want to
+  # remove before filtering `maf_df`
+  maf_df <- maf_df[, -16]
```

Comment:
> * Is this a new column or did we see this in other releases?

@jaclyn-taroni, this duplicate column was added to the `00-map-to-sample_id.R` script [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/6bdd22b504016e09df0d2801d79c34ddd6d83026/analyses/oncoprint-landscape/00-map-to-sample_id.R#L176) in merged PR #243 (https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/243#discussion_r345794392). 

My understanding is that the original `Tumor_Sample_Barcode` column in the maf file is needed to filter to tumor samples only as this column holds the `Kids_First_Biospecimen_ID` values https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/6bdd22b504016e09df0d2801d79c34ddd6d83026/analyses/oncoprint-landscape/00-map-to-sample_id.R#L127

However, because we use `sample_id` values in the `01-plot-oncoprint.R` script, the `Tumor_Sample_Barcode` column was recoded to hold the `sample_id` values
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/6bdd22b504016e09df0d2801d79c34ddd6d83026/analyses/oncoprint-landscape/00-map-to-sample_id.R#L176

which resulted in a duplicated `Tumor_Sample_Barcode` column.

I initially tried to remove the original `Tumor_Sample_Barcode` column with the biospecimen IDs in the `00` script using:

`dplyr::select(-Tumor_Sample_Barcode) %>%` before the `rename` step linked above, but this initially produced an error for me, that I was interestingly unable to reproduce this morning so this may be a better solution here?

Note that it does not tell us it's duplicated, both columns are named `Tumor_Sample_Barcode`.

Reply:
> `dplyr::select(-Tumor_Sample_Barcode) %>%` before the `rename` step linked above, but this initially produced an error for me, that I was interestingly unable to reproduce this morning so this may be a better solution here?

My understanding is as follows: we want to replace the biospecimen IDs in the `Tumor_Sample_Barcode` column with `sample_id` so that we can match between the experimental strategies, _but_ we need to call the `sample_id` column `Tumor_Sample_Barcode` because that's a requirement of `oncoplot()`. That makes me think that that what you lay out that previously gave you an error is a good move!